examples/jsm/nodes/core/InputNode.js

```diff
@@ -1,80 +1,113 @@
+import { Color, Matrix3, Matrix4, Vector2, Vector3, Vector4 } from 'three';
 import Node from './Node.js';
 
-class InputNode extends Node {
+function getValueType( value ) {
 
-	constructor( inputType ) {
+	if ( typeof value === 'number' ) {
 
-		super( inputType );
+		return 'float';
 
-		this.inputType = inputType;
+	} else if ( typeof value === 'boolean' ) {
 
-		this.constant = false;
+		return 'bool';
 
-	}
+	} else if ( value?.isVector2 === true ) {
 
-	setConst( value ) {
+		return 'vec2';
 
-		this.constant = value;
+	} else if ( value?.isVector3 === true ) {
 
-		return this;
+		return 'vec3';
 
-	}
+	} else if ( value?.isVector4 === true ) {
 
-	getConst() {
+		return 'vec4';
 
-		return this.constant;
+	} else if ( value?.isMatrix3 === true ) {
 
-	}
+		return 'mat3';
 
-	getInputType( /* builder */ ) {
+	} else if ( value?.isMatrix4 === true ) {
 
-		return this.inputType;
+		return 'mat4';
+
+	} else if ( value?.isColor === true ) {
+
+		return 'color';
 
 	}
 
-	getInputHash( builder ) {
+}
 
-		return this.getHash( builder );
+function getValueFromType( type ) {
 
-	}
+	if ( type === 'color' ) {
 
-	generateConst( builder ) {
+		return new Color();
 
-		return builder.getConst( this.getNodeType( builder ), this.value );
+	} else if ( type.includes( 'vec2' ) ) {
 
-	}
+		return new Vector2();
 
-	generate( builder, output ) {
+	} else if ( type.includes( 'vec3' ) ) {
```

Comment:
I think that `type === 'vec3'` is more optimized.

Reply:
This also includes 'bvec3', 'ivec3', and 'uvec3'. I can change it to `type.slice( -3 ) === 'vec3'`.

Reply:
Oh okay. let's keep it that way 👍

Reply:
Done.